walkthroughs/howto-k8s-tls-acm/README.md

```diff
@@ -0,0 +1,221 @@
+    --certificate-authority-arn $ROOT_CA_ARN \
+    --certificate "${ROOT_CA_CERT}"
+```
+
+We also want to grant permissions to the CA to automatically renew the managed certificates it issues:
+
+```bash
+aws acm-pca create-permission \
+    --certificate-authority-arn $ROOT_CA_ARN \
+    --actions IssueCertificate GetCertificate ListPermissions \
+    --principal acm.amazonaws.com
+```
+
+Request a managed certificate from ACM using this CA:
+
+```bash
+export CERTIFICATE_ARN=`aws acm request-certificate \
+    --domain-name "*.${SERVICES_DOMAIN}" \
+    --certificate-authority-arn ${ROOT_CA_ARN} \
+    --query CertificateArn --output text`
+```
+
+With managed certificates, ACM will automatically renew certificates that are nearing the end of their validity, and App Mesh will automatically distribute the renewed certificates. See [Managed Renewal and Deployment](https://aws.amazon.com/certificate-manager/faqs/#Managed_Renewal_and_Deployment) for more information.
+
+## Step 3: Create a Mesh with TLS enabled
+
+We are going to setup a mesh with four virtual nodes: Frontend, Blue, Green and Red, one virtual service: color and one virtual router: color.
+
+Let's create the mesh.
+
+```bash
+./mesh.sh up
+```
+
+Frontend has backend virtual service (color) configured and the virtual service (color) uses virtual router (color) as the provider. The virtual router (color) has three routes configured:
+- color-route-blue: matches on HTTP header "blue" to route traffic to virtual node `blue`
+- color-route-green: matches on HTTP header "green" to route traffic to virtual node `green` 
+- color-route-red: matches on HTTP header "red" to route traffic to virtual node `red`
+
+Virtual node `blue` is configured with TLS enabled for it's respective listeners. Here's the spec for `blue` Virtual Node:
+
+```
+apiVersion: appmesh.k8s.aws/v1beta2
+kind: VirtualNode
+metadata:
+  name: blue
+  namespace: howto-k8s-tls-acm
+spec:
+  podSelector:
+    matchLabels:
+      app: color
+      version: blue
+  listeners:
+    - portMapping:
+        port: 8080
+        protocol: http
+      healthCheck:
+        protocol: http
+        path: '/ping'
+        healthyThreshold: 2
+        unhealthyThreshold: 2
+        timeoutMillis: 2000
+        intervalMillis: 5000
+      tls:
+        mode: STRICT
+        certificate:
+          acm:
+            certificateARN: arn:aws:acm:eu-west-2:<ACCOUNT_ID>>:certificate/<certificate>
+  serviceDiscovery:
+    dns:
+      hostname: color-blue.howto-k8s-tls-acm.svc.cluster.local
+```
+
+The `tls` block specifies the ACM certificate to use. See [Transport Layer Security (TLS)](https://docs.aws.amazon.com/app-mesh/latest/userguide/tls.html) for the IAM permissions that are required while using ACM PCA for TLS.
+
+
+## Setup 4: Verify TLS is enabled
+
+```bash
+kubectl -n default run -it --rm curler --image=tutum/curl /bin/bash
+```
+
+```
+curl -H "color_header: blue" front.howto-k8s-tls-acm.svc.cluster.local:8080/; echo;
```

Comment:
Thanks for the work so far, this looks great. I got as far in testing this as here, and my curls failed.

Looking at the controller logs, I'm seeing:

"ForbiddenException: Call requires the acm:DescribeCertificate permission for the certificate with the ARN arn:aws:acm:..."

This is the first time I've used this with EKS, let's double check what permissions are needed. I'll check into this tomorrow sometime, but if you can beat me to it, please go for it.

Reply:
It looks like adding AWSAppMeshFullAccess will work, it includes acm-pca:DescribeCertificateAuthority. Can you add this in a note toward the beginning, and then test the walkthrough through each step?

Reply:
I added a note regarding the IAM permissions required to use ACM PCA for TLS in the beginning of the Steps. I have provided a link to the AWS document for verifying the IAM permissions in order to incorporate any additional permissions added to the AWS documentation in future. 
I have also tested the walkthrough after making this change.

Reply:
Cool, I worked through the updated walkthrough. I think that note could be a bit more user-friendly, maybe just a quick sentence or two about how to add the required permissions?

So that's resolved the controller permissions issue, but I'm now getting a 503 when calling through front. From logs:

Trying to hit color.howto-k8s-tls-acm.svc.cluster.local
10.100.107.221
[ERROR] HTTP Error 503: Service Unavailable
127.0.0.1 - - [16/Sep/2020 14:39:22] code 503, message Service Unavailable
127.0.0.1 - - [16/Sep/2020 14:39:22] "GET / HTTP/1.1" 503 -

Getting this response for any color.

So I think two additional things needed for this walkthrough:

* The controller install should follow the instructions in https://github.com/aws/eks-charts/tree/master/stable/appmesh-controller#eks-with-iam-roles-for-service-account

* It seems the node instance role needs both "acm:ExportCertificate" and "acm-pca:GetCertificateAuthorityCertificate"

With all of this in place, I can make this work. Maybe you already had these configured in your cluster? With a walkthrough like this, it's sometimes good to start with a fresh cluster so you can test the user experience.

If you can fix these up I can give it a final walkthrough.

Reply:
I have added more details regarding the IAM permissions and explictly mentioned the additional IAM permission in the walkthrough that are required for ACM PCA for TLS. The cleanup of appmesh-controller and namespace has also been added to the walkthrough.